Makefile-test.mk

```diff
@@ -55,7 +55,17 @@ endif
 
 # Folder where the e2e tests are located.
 E2E_TARGET ?= ./test/e2e/...
-E2E_KIND_VERSION ?= kindest/node:v1.32.0
+E2E_K8S_VERSION ?= 1.32
+ifeq ($(E2E_K8S_VERSION),1.30)
+	E2E_K8S_VERSION_PATCH ?= 10
+else ifeq ($(E2E_K8S_VERSION),1.31)
+	E2E_K8S_VERSION_PATCH ?= 6
+else ifeq ($(E2E_K8S_VERSION),1.32)
+	E2E_K8S_VERSION_PATCH ?= 3
+else
+	E2E_K8S_VERSION_PATCH ?= 0
+endif
+E2E_KIND_VERSION ?= kindest/node:v$(E2E_K8S_VERSION).$(E2E_K8S_VERSION_PATCH)
```

Comment:
This looks great to me, but I'm wondering what we should do with `E2E_K8S_VERSIONS`. Is it still necessary? 

It seems not even aligned with the block above. Maybe @mbobrovskyi knows what is `E2E_K8S_VERSIONS` used for as recently updating. I think the intention is to have one test per minor version, so properly no need to add patch to the versions.

Reply:
> It is not aligned in the PR even :) , what are the consequences of merging unaligned?
yes :D I didn't touch it because it was already unaligned with what `test-infra` sets (which is what this PR sets)

> sounds reasonable, but I would like to see the prototype to assess.
I could have another PR with this if it's finally needed, although @mbobrovskyi suggests is not? https://github.com/kubernetes-sigs/kueue/pull/4950#discussion_r2041785759